crypto/cmp/cmp_client.c

```diff
@@ -284,7 +284,7 @@ static int poll_for_response(OSSL_CMP_CTX *ctx, int sleep, int rid,
             if (check_after < 0 || (uint64_t)check_after
                 > (sleep ? ULONG_MAX / 1000 : INT_MAX)) {
                 CMPerr(0, CMP_R_CHECKAFTER_OUT_OF_RANGE);
-                if (BIO_snprintf(str, OSSL_CMP_PKISI_BUFLEN, "value = %ld",
+                if (BIO_snprintf(str, OSSL_CMP_PKISI_BUFLEN, "value = %jd",
```

Comment:
For clarity I think there should be cast `(intmax_t)check_after` although this just an alias.

Reply:
Yes, probably, I wonder how many other places need that.
OTOH, I use %ld mostly in my own code, because it is more portable,
and cast the value to (long) and live with the possible data loss.